### DISCUSSION TITLE
Installation of TTS

### DISCUSSION THREAD

**Discussion by Parvezkhan0** [coqui-ai/TTS#2962] [General Q&A]
While performing the command 'pip install TTS' , I got the following error - 

pip install TTS
Defaulting to user installation because normal site-packages is not writeable
Collecting TTS
  Using cached TTS-0.17.4.tar.gz (1.6 MB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [238 lines of output]
      Collecting setuptools
        Obtaining dependency information for setuptools from https://files.pythonhosted.org/packages/bb/26/7945080113158354380a12ce26873dd6c1ebd88d47f5bc24e2c5bb38c16a/setuptools-68.2.2-py3-none-any.whl.metadata
        Using cached setuptools-68.2.2-py3-none-any.whl.metadata (6.3 kB)
      Collecting wheel
        Obtaining dependency information for wheel from https://files.pythonhosted.org/packages/b8/8b/31273bf66016be6ad22bb7345c37ff350276cfd46e389a0c2ac5da9d9073/wheel-0.41.2-py3-none-any.whl.metadata
        Using cached wheel-0.41.2-py3-none-any.whl.metadata (2.2 kB)
      Collecting cython==0.29.30
        Using cached Cython-0.29.30-py2.py3-none-any.whl (985 kB)
      Collecting numpy==1.22.0
        Using cached numpy-1.22.0.zip (11.3 MB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'done'
        Preparing metadata (pyproject.toml): started
        Preparing metadata (pyproject.toml): finished with status 'done'
      Collecting packaging
        Using cached packaging-23.1-py3-none-any.whl (48 kB)
      Using cached setuptools-68.2.2-py3-none-any.whl (807 kB)
      Using cached wheel-0.41.2-py3-none-any.whl (64 kB)
      Building wheels for collected packages: numpy
        Building wheel for numpy (pyproject.toml): started
        Building wheel for numpy (pyproject.toml): finished with status 'error'
        error: subprocess-exited-with-error

        Building wheel for numpy (pyproject.toml) did not run successfully.
        exit code: 1

        [203 lines of output]
        setup.py:66: RuntimeWarning: NumPy 1.22.0 may not yet support Python 3.11.
          warnings.warn(
        Running from numpy source directory.
        Processing numpy/random\_bounded_integers.pxd.in
        Processing numpy/random\bit_generator.pyx
        Processing numpy/random\mtrand.pyx
        Processing numpy/random\_bounded_integers.pyx.in
        Processing numpy/random\_common.pyx
        Processing numpy/random\_generator.pyx
        Processing numpy/random\_mt19937.pyx
        Processing numpy/random\_pcg64.pyx
        Processing numpy/random\_philox.pyx
        Processing numpy/random\_sfc64.pyx
        Cythonizing sources
        INFO: blas_opt_info:
        INFO: blas_armpl_info:
        INFO: No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
        INFO: customize MSVCCompiler
        INFO:   libraries armpl_lp64_mp not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: blas_mkl_info:
        INFO:   libraries mkl_rt not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: blis_info:
        INFO:   libraries blis not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: openblas_info:
        INFO:   libraries openblas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO: get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'
        INFO: customize GnuFCompiler
        WARN: Could not locate executable g77
        WARN: Could not locate executable f77
        INFO: customize IntelVisualFCompiler
        WARN: Could not locate executable ifort
        WARN: Could not locate executable ifl
        INFO: customize AbsoftFCompiler
        WARN: Could not locate executable f90
        INFO: customize CompaqVisualFCompiler
        INFO: Found executable C:\Program Files\Git\usr\bin\DF.exe
        WARN: Could not locate executable C:\Program
        INFO: customize IntelItaniumVisualFCompiler
        WARN: Could not locate executable efl
        INFO: customize Gnu95FCompiler
        WARN: Could not locate executable gfortran
        WARN: Could not locate executable f95
        INFO: customize G95FCompiler
        WARN: Could not locate executable g95
        INFO: customize IntelEM64VisualFCompiler
        INFO: customize IntelEM64TFCompiler
        WARN: Could not locate executable efort
        WARN: Could not locate executable efc
        INFO: customize PGroupFlangCompiler
        WARN: Could not locate executable flang
        WARN: don't know how to compile Fortran code on platform 'nt'
        INFO:   NOT AVAILABLE
        INFO:
        INFO: accelerate_info:
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_3_10_blas_threads_info:
        INFO: Setting PTATLAS=ATLAS
        INFO:   libraries tatlas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_3_10_blas_info:
        INFO:   libraries satlas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_blas_threads_info:
        INFO: Setting PTATLAS=ATLAS
        INFO:   libraries ptf77blas,ptcblas,atlas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_blas_info:
        INFO:   libraries f77blas,cblas,atlas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-install-xak73vr7\numpy_5a2feafee9274bd5a5fdd6bac7c4c1fb\numpy\distutils\system_info.py:2077: UserWarning:
            Optimized (vendor) Blas libraries are not found.
            Falls back to netlib Blas library which has worse performance.
            A better performance should be easily gained by switching
            Blas library.
          if self._calc_info(blas):
        INFO: blas_info:
        INFO:   libraries blas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-install-xak73vr7\numpy_5a2feafee9274bd5a5fdd6bac7c4c1fb\numpy\distutils\system_info.py:2077: UserWarning:
            Blas (http://www.netlib.org/blas/) libraries not found.
            Directories to search for the libraries can be specified in the
            numpy/distutils/site.cfg file (section [blas]) or by setting
            the BLAS environment variable.
          if self._calc_info(blas):
        INFO: blas_src_info:
        INFO:   NOT AVAILABLE
        INFO:
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-install-xak73vr7\numpy_5a2feafee9274bd5a5fdd6bac7c4c1fb\numpy\distutils\system_info.py:2077: UserWarning:
            Blas (http://www.netlib.org/blas/) sources not found.
            Directories to search for the sources can be specified in the
            numpy/distutils/site.cfg file (section [blas_src]) or by setting
            the BLAS_SRC environment variable.
          if self._calc_info(blas):
        INFO:   NOT AVAILABLE
        INFO:
        non-existing path in 'numpy\\distutils': 'site.cfg'
        INFO: lapack_opt_info:
        INFO: lapack_armpl_info:
        INFO:   libraries armpl_lp64_mp not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: lapack_mkl_info:
        INFO:   libraries mkl_rt not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: openblas_lapack_info:
        INFO:   libraries openblas not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: openblas_clapack_info:
        INFO:   libraries openblas,lapack not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: flame_info:
        INFO:   libraries flame not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_3_10_threads_info:
        INFO: Setting PTATLAS=ATLAS
        INFO:   libraries tatlas,tatlas not found in C:\Program Files\Python311\lib
        INFO:   libraries tatlas,tatlas not found in C:\
        INFO:   libraries tatlas,tatlas not found in C:\Program Files\Python311\libs
        INFO: <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_3_10_info:
        INFO:   libraries satlas,satlas not found in C:\Program Files\Python311\lib
        INFO:   libraries satlas,satlas not found in C:\
        INFO:   libraries satlas,satlas not found in C:\Program Files\Python311\libs
        INFO: <class 'numpy.distutils.system_info.atlas_3_10_info'>
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_threads_info:
        INFO: Setting PTATLAS=ATLAS
        INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\Program Files\Python311\lib
        INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\
        INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\Program Files\Python311\libs
        INFO: <class 'numpy.distutils.system_info.atlas_threads_info'>
        INFO:   NOT AVAILABLE
        INFO:
        INFO: atlas_info:
        INFO:   libraries f77blas,cblas,atlas not found in C:\Program Files\Python311\lib
        INFO:   libraries f77blas,cblas,atlas not found in C:\
        INFO:   libraries f77blas,cblas,atlas not found in C:\Program Files\Python311\libs
        INFO: <class 'numpy.distutils.system_info.atlas_info'>
        INFO:   NOT AVAILABLE
        INFO:
        INFO: lapack_info:
        INFO:   libraries lapack not found in ['C:\\Program Files\\Python311\\lib', 'C:\\', 'C:\\Program Files\\Python311\\libs']
        INFO:   NOT AVAILABLE
        INFO:
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-install-xak73vr7\numpy_5a2feafee9274bd5a5fdd6bac7c4c1fb\numpy\distutils\system_info.py:1902: UserWarning:
            Lapack (http://www.netlib.org/lapack/) libraries not found.
            Directories to search for the libraries can be specified in the
            numpy/distutils/site.cfg file (section [lapack]) or by setting
            the LAPACK environment variable.
          return getattr(self, '_calc_info_{}'.format(name))()
        INFO: lapack_src_info:
        INFO:   NOT AVAILABLE
        INFO:
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-install-xak73vr7\numpy_5a2feafee9274bd5a5fdd6bac7c4c1fb\numpy\distutils\system_info.py:1902: UserWarning:
            Lapack (http://www.netlib.org/lapack/) sources not found.
            Directories to search for the sources can be specified in the
            numpy/distutils/site.cfg file (section [lapack_src]) or by setting
            the LAPACK_SRC environment variable.
          return getattr(self, '_calc_info_{}'.format(name))()
        INFO:   NOT AVAILABLE
        INFO:
        INFO: numpy_linalg_lapack_lite:
        INFO:   FOUND:
        INFO:     language = c
        INFO:     define_macros = [('HAVE_BLAS_ILP64', None), ('BLAS_SYMBOL_SUFFIX', '64_')]
        INFO:
        Warning: attempted relative import with no known parent package
        C:\Users\Parvez Khan\AppData\Local\Temp\pip-build-env-ecrm0hdu\overlay\Lib\site-packages\setuptools\_distutils\dist.py:275: UserWarning: Unknown distribution option: 'define_macros'
          warnings.warn(msg)
        running bdist_wheel
        running build
        running config_cc
        INFO: unifing config_cc, config, build_clib, build_ext, build commands --compiler options
        running config_fc
        INFO: unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
        running build_src
        INFO: build_src
        INFO: building py_modules sources
        creating build
        creating build\src.win-amd64-3.11
        creating build\src.win-amd64-3.11\numpy
        creating build\src.win-amd64-3.11\numpy\distutils
        INFO: building library "npymath" sources
        error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
        [end of output]

        note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for numpy
      Failed to build numpy
      ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by Shaon2221** [comment]
Go to this link and download Microsoft C++ Build Tools:
https://visualstudio.microsoft.com/visual-cpp-build-tools/

![DKWVM](https://github.com/coqui-ai/TTS/assets/18596945/f08cacab-59bc-4a24-b064-cd5d01505c04)

**Reply by juandsep** [reply]
Same error, works for me. Thanks so much.

**Reply by mattkoczwara** [reply]
Is CUDA v11 still not supported?

**Comment by milobug** [comment]
非常感谢,有用

**Comment by decimeter001** [comment]
问题解决了，非常感谢
